### PR TITLE
trilinos: update to 12.4.2; fix hypre options and mumps libraries order

### PR DESCRIPTION
**Pull request by davydden**
superseeds https://github.com/Homebrew/homebrew-science/pull/2930

### CODE REVIEW

**Comment by dpo** [comment]
https://github.com/Homebrew/homebrew/pull/46461

File: trilinos.rb [hunk 1]
```diff
@@ -58,7 +57,7 @@ class Trilinos < Formula
 
   # Experimental TPLs:
   depends_on "eigen"        => :recommended
-  depends_on "hypre"        => [:recommended] + mpidep + openblasdep # EpetraExt tests fail to compile
+  depends_on "hypre"        => [:recommended] + ((build.with? "mpi") ? [] : ["without-mpi"]) + openblasdep # EpetraExt tests fail to compile
```
Comment (dpo) [inline trilinos.rb]: Could you check whether this can be restored, now that the MPI issues have been fixed?

Comment (davydden) [inline trilinos.rb]: it's not related to mpi issues, AFAIK. The problem was that `hypre` does not have `--with-mpi`, it has `--without-mpi`.

Comment (dpo) [inline trilinos.rb]: Ah, you're right.

**Comment by dpo** [comment]
@BrewTestBot test this please

**Comment by dpo** [comment]
It times out. We need two hours of build time...

**Comment by dpo** [comment]
@mikemcquaid Is there any chance we could request extra build time for specific formulae? This one is particularly tricky and there's no way it will fit in 60 minutes.

**Comment by mikemcquaid** [comment]
@dpo I'll try and see if I can find a time when the CI isn't too busy. In general though: unfortunately not.

**Comment by mikemcquaid** [comment]
@BrewTestBot test this please

**Comment by mikemcquaid** [comment]
@dpo I've bumped this to 120m and scheduled a rebuild.

**Comment by dpo** [comment]
@mikemcquaid Thanks! That should do it.

**Comment by davydden** [comment]
i could be wrong, but it seems the build was not restarted, or maybe it's still in the queue...

**Comment by dpo** [comment]
It seems to have worked.

**Pull request closed without merging**